pytest/tests/sanity/state_sync_routed.py

```diff
@@ -0,0 +1,122 @@
+# Spins two block producers and two observers.
+# Wait several epochs and spin up another observer that
+# is blacklisted by both block producers.
+#
+# Make sure the new observer sync via routing through other observers.
+#
+# Three modes:
+#   - notx: no transactions are sent, just checks that
+#     the second node starts and catches up
+#   - onetx: sends one series of txs at the beginning,
+#     makes sure the second node balances reflect them
+#   - manytx: constantly issues txs throughout the test
+#     makes sure the balances are correct at the end
+
+import sys, time
+
+sys.path.append('lib')
+
+if len(sys.argv) < 3:
+    print("python state_sync.py [notx, onetx, manytx] <launch_at_block>")
+    exit(1)
+
+mode = sys.argv[1]
+assert mode in ['notx', 'onetx', 'manytx']
+
+from cluster import init_cluster, spin_up_node, load_config
+from utils import TxContext, LogTracker
+
+START_AT_BLOCK = int(sys.argv[2])
+TIMEOUT = 150 + START_AT_BLOCK * 10
+
+config = load_config()
+
+near_root, node_dirs = init_cluster(2, 3, 1, config, [["min_gas_price", 0], ["max_inflation_rate", 0], ["epoch_length", 10], ["block_producer_kickout_threshold", 80]], {u: {"tracked_shards": [0]} for u in range(2, 5)})
```

Comment:
only the last observer should track the shard.
if second and first observers also track the shard, the third observer will just state sync from them instead of using routing.

Reply:
fixed